### DISCUSSION TITLE
Specific Documentation

### DISCUSSION THREAD

**Discussion by evantrimboli** [mikro-orm/mikro-orm#1900] [Ideas]
Is it possible to annotate parts of the API documentation?

As an example, the `init` method on `Collection` returns a `Collection` instance. However it's unclear without looking at the source that it returns itself. This kind of detail doesn't seem to be something that belongs under the "docs" section on the site.

Do you have any thoughts on where would be best to make notes about these things?

**Comment by B4nan** [comment]
the signature is literally `async init(options?: InitOptions<T>): Promise<this>`, so it seems quite easy to understand what it does...

**Reply by evantrimboli** [reply]
The api docs list the return type as 

Returns: `Promise<Collection<T, O>>`

The same thing in vs code, there isn't an indication that it returns `this.

![image](https://user-images.githubusercontent.com/197184/120719618-ef999a80-c50d-11eb-8cee-0f4cbe016646.png)

**Reply by B4nan** [reply]
don't blame me for how TS works :] sure we can add jsdoc saying it provides fluent interface
